### ISSUE TITLE
Cannot read property 'api_key' of undefined 

### ISSUE THREAD

**Issue by maxwellpopescu** [nuxt-community/cloudinary-module#31]
As per the first example in [https://cloudinary.nuxtjs.org/usage/advanced](https://cloudinary.nuxtjs.org/usage/advanced), I am trying to upload images in my assets folder before Nuxt builds.

**nuxt.config.js**:

```
modules: [
  '@nuxtjs/cloudinary',
],
cloudinary: {
    cloudName: 'mycloudname',
    apiKey: 'myapikey',
    apiSecret: 'myapisecret',
},
hooks: {
  build: {
    before() {
      const { $cloudinary } = require('@nuxtjs/cloudinary')

      const assetsFolder = path.join(__dirname, 'assets/images')

      fs.readdir(assetsFolder, async (err, files) => {
        if (err) return

        /* Upload to Cloudinary */
        const uploadedAssets = await Promise.all(
          files.map((file) =>
            $cloudinary.upload(path.join(assetsFolder, file))
          )
        )
        /* Do something with the assets' info returned */
        console.log(uploadedAssets)
      })
    },
  },
},
```

The only difference between the demo code and my code is that I have defined the apiKey and apiSecret in the cloudinary options. However, I get the following error:

```
 ERROR  Cannot read property 'api_key' of undefined                                        19:52:32

  at CloudinaryApi.upload (node_modules/@nuxtjs/cloudinary/lib/templates/cloudinary-api.js:138:18)
  at nuxt.config.js:138:42
  at Array.map (<anonymous>)
  at nuxt.config.js:138:19
  at FSReqCallback.oncomplete (fs.js:156:23)
```

**Comment by maxwellpopescu** [comment]
The API defines options as an optional parameter to the upload function, but accesses `options.api_key`. 

**cloudinary-api.js**
```
  /**
   * Returns a Promise from the upload process for an asset on Server side
   * @param {String} file - path to the asset you wish to upload 
   * @param {Options} options - upload options (optional): https://cloudinary.com/documentation/image_upload_api_reference#optional_parameters
   * @param {Function} callback - callback handler (optional)
   * @returns {Promise<Asset> | Promise<{ msg: {String} }>}
   */
  upload(file, options, callback) {
    if (!options.api_key && !this._configurations.api_key
      && !options.api_secret && !this._configurations.api_secret) {
      throw new Error('API Key and Secret Key are needed for upload API')
    }
```

Including a dummy options parameter when calling the upload function fixes this, but the check for API key & secret should be changed so an error isn't thrown when `options` is not passed.

**Comment by mayashavin** [comment]
@maxwellpopescu Thanks a lot for your feedback :). Indeed it is a bug here. I have made the PR for fixing this issue. Expected to be available on v1.0.1 :)
